### PR TITLE
Install texture compression presets from json files in the project

### PR DESCRIPTION
**Pull request by Jhonnyg**
Part one of supporting custom texture compression presets from projects. You can now add a "my_file.texc_json" file to your project with settings for a texture compressor in JSON format:

```json
{
    "name": "BasisU",
    "presets": [
        {
            "name": "BASISU_CUSTOM_PRESET",
            "args": {
                "rdo_uastc": 0,
                "pack_uastc_flags": 2,
                "rdo_uastc_quality_scalar": 0.1
            }
        }
    ]
}
```

The second part of this feature is to adjust the source format for the texture profiles to actually being able to specify which preset to use for a specific compressor.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/AbstractProtoBuilderTest.java [hunk 1]
```diff
@@ -152,7 +151,7 @@ protected static Object getField(Class<?> klass, Object instance, String fieldNa
         }
     }
 
-    protected Project GetProject() {
+    protected Project getProject() {
```
Comment (Jhonnyg) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/AbstractProtoBuilderTest.java:154]: just a small nitpick

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java [hunk 2]
```diff
@@ -145,7 +138,7 @@ public enum OutputFlags {
     private List<URL> libUrls = new ArrayList<URL>();
     private List<String> propertyFiles = new ArrayList<>();
     private List<String> buildServerHeaders = new ArrayList<>();
-    private List<String> excluedFilesAndFoldersEntries = new ArrayList<>();
+    private List<String> excludedFilesAndFoldersEntries = new ArrayList<>();
```
Comment (Jhonnyg) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:141]: typo

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java [hunk 3]
```diff
@@ -35,18 +36,7 @@
...
-import java.util.Collections;
-import java.util.Enumeration;
-import java.util.EnumSet;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (Jhonnyg) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:39]: Hm, intellij decided this was a good idea 🤔

Comment (JCash) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:39]: Yeah, it's so annoying. SUrely including _fewer_ files is better for everyone?

**Review by JCash — DISMISSED** [review]
lgtm!

**Pull request merged**